modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CppPistacheServerCodegen.java

```diff
@@ -267,61 +269,62 @@ public OperationsMap postProcessOperationsWithModels(OperationsMap objs, List<Mo
-                if (param.isFile) isParsingSupported = false;
-                if (param.isCookieParam) isParsingSupported = false;
+        boolean consumeJson = false;
+        if (op.consumes != null) {
+            Predicate<Map<String,String>> isMediaTypeJson =  consume -> (consume.get("mediaType") != null && consume.get("mediaType").equals("application/json"));
+            consumeJson =  op.consumes.stream().anyMatch(isMediaTypeJson);
+        }
+        op.vendorExtensions.put("x-codegen-pistache-consumes-json", consumeJson);
 
-                //TODO: This changes the info about the real type but it is needed to parse the header params
-                if (param.isHeaderParam) {
-                    param.dataType = "std::optional<Pistache::Http::Header::Raw>";
-                    param.baseType = "std::optional<Pistache::Http::Header::Raw>";
-                } else if (param.isQueryParam) {
-                    if (param.isPrimitiveType) {
-                        param.dataType = "std::optional<" + param.dataType + ">";
-                    } else {
-                        param.dataType = "std::optional<" + param.dataType + ">";
-                        param.baseType = "std::optional<" + param.baseType + ">";
-                    }
-                }
-            }
+        op.httpMethod = op.httpMethod.substring(0, 1).toUpperCase(Locale.ROOT) + op.httpMethod.substring(1).toLowerCase(Locale.ROOT);
 
-            if (op.vendorExtensions == null) {
-                op.vendorExtensions = new HashMap<>();
-            }
-            op.vendorExtensions.put("x-codegen-pistache-consumes-json", consumeJson);
-            op.vendorExtensions.put("x-codegen-pistache-is-parsing-supported", isParsingSupported);
+        boolean isParsingSupported = true;
+        for (CodegenParameter param : op.allParams) {
+            boolean paramSupportsParsing =  (!param.isFormParam && !param.isFile && !param.isCookieParam);
+            isParsingSupported = isParsingSupported && paramSupportsParsing;
 
-            // Check if any one of the operations needs a model, then at API file level, at least one model has to be included.
-            for (String hdr : op.imports) {
-                if (importMapping.containsKey(hdr)) {
-                    continue;
-                }
-                operations.put("hasModelImport", true);
-            }
+            postProcessSingleParam(param);
         }
+        op.vendorExtensions.put("x-codegen-pistache-is-parsing-supported", isParsingSupported);
 
-        return objs;
+        // Check if any one of the operations needs a model, then at API file level, at least one model has to be included.
+        Predicate<String> importNotInImportMapping = hdr -> !importMapping.containsKey(hdr);
+        if (op.imports.stream().anyMatch(importNotInImportMapping)) {
+            operations.put("hasModelImport", true);
+        }
+    }
+
+    private static void postProcessSingleParam(CodegenParameter param) {
```

Comment:
thanks for the PR. 

Minor suggestion: what about adding a 1-liner docstring explaining what this function does?

Reply:
Yep, I'll add the suggested docstring.

Reply:
Documented with https://github.com/OpenAPITools/openapi-generator/pull/15279/commits/adf3e97e928f2ee7ddece576fbab9f6e8d21fd9c